src/components/orders/OrderPriceDisplay/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import styled from 'styled-components'
+import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+import { formatSmart, safeTokenName, TokenErc20 } from '@gnosis.pm/dex-js'
+
+import { getOrderExecutedPrice, getOrderLimitPrice, RawOrder } from 'api/operator'
+
+const Wrapper = styled.span`
+  display: flex;
+  align-items: center;
+  font-size: ${({ theme }): string => theme.fontSizeNormal};
+`
+
+const Icon = styled(FontAwesomeIcon)`
+  background: ${({ theme }): string => theme.textSecondary2}33; /* 33==20% transparency in hex */
+  border-radius: 1rem;
+  width: 2rem !important; /* FontAwesome sets it to 1em with higher specificity */
+  height: 2rem;
+  padding: 0.4rem;
+  margin-left: 0.5rem;
+`
+
+export type Props = {
```

Comment:
Are we sure we want this signature for the proos? It feels weird to hand in an order and a type. 
why not to hand in a fraction: numerator, denominator and allow the component to display any prices (not just order prices)

Will this work with fills? they also have execution prices? it feels a bit rigid. Although maybe for this ones, it will be another component, because the invert icon is not in it... (or maybe it is and is just to not show it, not sure)

Reply:
I agree with your concerns.

I ended up with this mainly for 2 reasons:
1. To encapsulate the price inversion inside the component, to avoid parent re-renders
2. I already had implemented the utils fns that take a parameter to invert the price, thus the `type: 'limit' | 'executed'`

So, the implementation got constrained by those decisions.

I'll rewrite this be more generic, and invert the price without relying on the nice (but useless) invert price param utils fn:

```typescript
type Props = {
  buyAmount: string
  buyToken: TokenErc20
  sellAmount: string
  sellToken: TokenErc20
  priceInverted?: boolean  // in case we want to control from the outside, like on the fills
  showInvertIcon?: boolean  // in case we don't want to show the invert button, like on the fills
}
```